src/components/main-content.tsx

```diff
@@ -22,13 +27,16 @@ export class MainContentComponent extends BaseComponent<IProps, IState> {
   }
 
   private renderDialog() {
-    const {ui} = this.stores;
-    const {showInvestigationModalSelect} = this.stores.ui;
+    const {showInvestigationModalSelect, investigationPanelSpace} = this.stores.ui;
     const {showInvestigationPanel} = this.stores.ui;
+
+    // stawman code
+    const SpaceComponent = SpaceComponents[investigationPanelSpace];
+
     if (showInvestigationModalSelect) {
       return <InvestigateDialogComponent/>;
-    } else if (showInvestigationPanel) {
-      return <TwoUpDisplayComponent/>;
+    } else if (showInvestigationPanel && SpaceComponent) {
```

Comment:
If one of the values of type `Spaces` in the ui store is `none` then eventually we can probably whack `showInvestigationPanel` and instead just check if the current space is none (unless you intend none to have another meaning).

Reply:
Right.  We don't have clarity on this.  What is displayed when we first bring up the app?  An empty panel?  An intro panel?  One of the spaces?  I liked including `none` since it more resembles a C family enum where `none` represents "no spaces shown", but if that is never a valid option.... Agree on close button too.  I suppose we need to ask the right person if we can ever have a state where no spaces are shown.